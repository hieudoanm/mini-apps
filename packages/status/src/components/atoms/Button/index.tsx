import React from 'react';

export type ButtonProps = {
  type: 'button' | 'submit' | 'reset';
  onClick: () => void;
  children: string;
};

const Button: React.FC<ButtonProps> = ({ type, onClick, children }) => {
  return (
    <button
      type={type}
      onClick={onClick}
      className="rounded bg-red-500 text-white px-2 py-1 text-sm"
    >
      {children}
    </button>
  );
};

Button.displayName = 'Button';

export default Button;
