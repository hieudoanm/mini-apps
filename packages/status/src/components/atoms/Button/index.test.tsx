import { render } from '@testing-library/react';
import Button from '.';

describe('Button', () => {
  test('render default', () => {
    const wrapper = render(
      <Button type="button" onClick={() => {}}>
        Button
      </Button>
    );
    expect(wrapper.container).toMatchSnapshot();
  });
});
