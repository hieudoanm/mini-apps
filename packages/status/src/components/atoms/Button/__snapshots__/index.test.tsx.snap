// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render default 1`] = `
<div>
  <button
    class="rounded bg-red-500 text-white px-2 py-1 text-sm"
    type="button"
  >
    Button
  </button>
</div>
`;
